spec/connection.md

```diff
@@ -0,0 +1,229 @@
+# NATS Connection
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2023-10-12 |
+|Author  |@Jarema |
+|Status  |`Implemented`|
+|Tags    |client, server|
+
+|Revision|Date|Author|Info|
+ |--------|----|------|----|
+ |1       |2023-10-12|@author|Initial draft|
+
+## Summary
+
+This document describes how clients connect to the NATS server or NATS cluster.
+That includes topics like:
+- connection process
+- reconnect
+- tls
+- discoverability of other nodes in a cluster
+
+## Motivation
+
+Ensuring a consistent way how Clients establish and maintain connection with the NATS server and provide consistent and predictable behaviour across the ecosystem.
+
+## Guide-level Explanation
+
+### Establishing the connection
+
+#### Minimal example
+
+1. Clients initiate a TCP connection to the Server.
+2. Server responds with [INFO][INFO] json.
+3. Client sends [CONNECT][CONNECT] json.
```

Comment:
Good to get these in this doc so we have one way to know and reference all related information?

Reply:
The intention here was to avoid doc drift and always refer automatically generated and updated schemas from docs.

Reply:
Ah I didn’t realise those are auto generated. Probsbly good then